Index: yaird-0.0.11/perl/Blacklist.pm
===================================================================
--- yaird-0.0.11.orig/perl/Blacklist.pm.orig	2005-08-07 22:57:40.000000000 +0200
+++ yaird-0.0.11.orig/perl/Blacklist.pm	2005-09-12 05:48:43.000000000 +0200
@@ -36,7 +36,7 @@
 	}
 	my $blackMap = {};
 	my $hotplugDir = Conf::get ('hotplug');
-	my $blackListName = "hotplugDir/blacklist";
+	my $blackListName = "$hotplugDir/blacklist";
 	if (-e $blackListName) {
 		# The blacklist does not have to exist:
 		# there are machines without hotplug,
