--- yaird-0.0.11.orig/perl/Parser.pm
+++ yaird-0.0.11/perl/Parser.pm
@@ -168,7 +168,7 @@
 		# these identifiers to have same name as reserved
 		# words in config language, so no need to quote.
 		#
-identifier :	/[A-Za-z][A-Za-z0-9_]*/
+identifier :	/[A-Za-z0-9][A-Za-z0-9_-]*/
 	      |	<error>
 
 
@@ -395,7 +395,7 @@
 		# these identifiers to have same name as reserved
 		# words in config language, so no need to quote.
 		#
-identifier :	/[A-Za-z][A-Za-z0-9_]*/
+identifier :	/[A-Za-z0-9][A-Za-z0-9_-]*/
 	      |	<error>
 
 
