#! /bin/sh -e 

. $(dirname $0)/DPATCH

@DPATCH@
--- kernel-source-2.6.11/include/asm-alpha/spinlock.h~    Thu Mar  3 13:25:19 2005
+++ kernel-source-2.6.11/include/asm-alpha/spinlock.h     Thu May  5 01:16:35 2005
@@ -198,4 +198,6 @@ static inline void _raw_read_unlock(rwlo
        : "m" (*lock) : "memory");
 }

+#define _raw_read_trylock(lock) generic_raw_read_trylock(lock)
+
 #endif /* _ALPHA_SPINLOCK_H */
