--- openais.orig/Makefile
+++ openais/Makefile
@@ -37,23 +37,8 @@
 INCLUDEDIR_SERVICE=/usr/include/openais/service
 MANDIR=/usr/share/man
 ETCDIR=/etc/ais
-LCRSODIR=/usr/libexec/lcrso
-ARCH=$(shell uname -p)
-
-ifeq (,$(findstring 64,$(ARCH)))
-LIBDIR=/usr/lib/openais
-else
-LIBDIR=/usr/lib64/openais
-endif
-ifeq (s390,$(ARCH))
+LCRSODIR=/usr/lib/openais/lcrso
 LIBDIR=/usr/lib/openais
-endif
-ifeq (s390x,$(ARCH))
-LIBDIR=/usr/lib64/openais
-endif
-ifeq (ia64,$(ARCH))
-LIBDIR=/usr/lib/openais
-endif
 
 all:
 	(cd lcr; echo ==== `pwd` ===; $(MAKE) all);
@@ -158,7 +143,7 @@
 	install -m 755 exec/libtotem_pg.a $(DESTDIR)$(LIBDIR)
 endif
 
-	echo $(LIBDIR) > $(DESTDIR)/etc/ld.so.conf.d/openais-$(ARCH).conf
+	echo $(LIBDIR) > $(DESTDIR)/etc/ld.so.conf.d/openais.conf
 
 	install -m 755 exec/*lcrso $(DESTDIR)$(LCRSODIR)
 
--- openais.orig/lcr/lcr_ifact.c
+++ openais/lcr/lcr_ifact.c
@@ -174,7 +174,7 @@
 		path_list[0] = strdup (cwd);
 		path_list_entries++;
 	}
-	path_list[path_list_entries++] = "/usr/libexec/lcrso";
+	path_list[path_list_entries++] = "/usr/lib/openais/lcrso";
 }
 
 static void ld_library_path_build (void)
