--- openais.orig/Makefile
+++ openais/Makefile
@@ -115,40 +115,38 @@
 	cp -a lib/libcfg.so.2 $(DESTDIR)$(LIBDIR)
 	cp -a exec/libtotem_pg.so.2 $(DESTDIR)$(LIBDIR)
 
-	install -m 755 lib/libais.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaAmf.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaClm.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaCkpt.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaEvt.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaLck.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaMsg.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libevs.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libcpg.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libcfg.so.2.* $(DESTDIR)$(LIBDIR)
-	install -m 755 exec/libtotem_pg.so.2.* $(DESTDIR)$(LIBDIR)
-
-ifneq "NO" "$(STATICLIBS)"
-	install -m 755 lib/libais.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaAmf.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaClm.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaCkpt.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaEvt.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaLck.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libSaMsg.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libevs.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libcpg.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libevs.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libcpg.a $(DESTDIR)$(LIBDIR)
-	install -m 755 lib/libcfg.a $(DESTDIR)$(LIBDIR)
-	install -m 755 exec/libtotem_pg.a $(DESTDIR)$(LIBDIR)
-endif
+	install -m 644 lib/libais.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaAmf.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaClm.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaCkpt.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaEvt.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaLck.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaMsg.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libevs.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libcpg.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libcfg.so.2.* $(DESTDIR)$(LIBDIR)
+	install -m 644 exec/libtotem_pg.so.2.* $(DESTDIR)$(LIBDIR)
+
+	install -m 644 lib/libais.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaAmf.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaClm.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaCkpt.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaEvt.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaLck.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libSaMsg.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libevs.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libcpg.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libevs.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libcpg.a $(DESTDIR)$(LIBDIR)
+	install -m 644 lib/libcfg.a $(DESTDIR)$(LIBDIR)
+	install -m 644 exec/libtotem_pg.a $(DESTDIR)$(LIBDIR)
 
 	echo $(LIBDIR) > $(DESTDIR)/etc/ld.so.conf.d/openais.conf
 
-	install -m 755 exec/*lcrso $(DESTDIR)$(LCRSODIR)
+	install -m 644 exec/*lcrso $(DESTDIR)$(LCRSODIR)
 
 	install -m 755 exec/aisexec $(DESTDIR)$(SBINDIR)
-	install -m 700 exec/keygen $(DESTDIR)$(SBINDIR)/ais-keygen
+	install -m 755 exec/keygen $(DESTDIR)$(SBINDIR)/ais-keygen
 	install -m 644 conf/openais.conf $(DESTDIR)$(ETCDIR)
 	install -m 644 conf/amf.conf $(DESTDIR)$(ETCDIR)
 
