--- drivers/usb/usb-ohci.c.orig	2003-04-16 15:42:46.000000000 +0200
+++ drivers/usb/usb-ohci.c	2003-04-16 15:45:41.000000000 +0200
@@ -490,12 +490,17 @@
 				usb_pipeout (urb->pipe)
 					? PCI_DMA_TODEVICE
 					: PCI_DMA_FROMDEVICE);
-			urb->complete (urb);
+			if (urb->interval) {
+				urb->complete (urb);
 
-			/* implicitly requeued */
-  			urb->actual_length = 0;
-			urb->status = -EINPROGRESS;
-			td_submit_urb (urb);
+				/* implicitly requeued */
+				urb->actual_length = 0;
+				urb->status = -EINPROGRESS;
+				td_submit_urb (urb);
+			} else {
+				urb_rm_priv(urb);
+				urb->complete (urb);
+			}
   			break;
   			
 		case PIPE_ISOCHRONOUS:
