--- ./include/asm-sparc/system.h.bad	Sat Sep 18 14:15:43 2004
+++ ./include/asm-sparc/system.h	Sat Feb 12 10:55:05 2005
@@ -295,9 +295,9 @@
 #define wmb()	mb()
 #define set_mb(__var, __value)  do { __var = __value; mb(); } while(0)
 #define set_wmb(__var, __value) set_mb(__var, __value)
-#define smp_mb()	__asm__ __volatile__("":::"memory");
-#define smp_rmb()	__asm__ __volatile__("":::"memory");
-#define smp_wmb()	__asm__ __volatile__("":::"memory");
+#define smp_mb()	__asm__ __volatile__("":::"memory")
+#define smp_rmb()	__asm__ __volatile__("":::"memory")
+#define smp_wmb()	__asm__ __volatile__("":::"memory")
 
 #define nop() __asm__ __volatile__ ("nop");
 
