diff -Naru a/arch/x86_64/ia32/ptrace32.c b/arch/x86_64/ia32/ptrace32.c
--- a/arch/x86_64/ia32/ptrace32.c	2005-08-16 21:49:44 -07:00
+++ b/arch/x86_64/ia32/ptrace32.c	2005-08-16 21:49:44 -07:00
@@ -185,12 +185,12 @@
 			goto out;
 		*err = ptrace_check_attach(child, request == PTRACE_KILL); 
 		if (*err < 0) 
-				goto out;
+			goto out;
 		return child; 
-	} 
 
  out:
-	free_task_struct(child);
+		free_task_struct(child);
+	} 
 	return NULL; 
 	
 } 
# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/06 10:51:33-02:00 mbellon@mvista.com 
#   [PATCH] 32 bit ltrace oops when tracing 64 bit executable [X86_64]
#   
#   Didn't see  a fix for this so here it is. Tried using "ltrace -i" on a
#   64 bit executable when ltrace was a 32 bit executable. The kernel threw
#   an oops.
#   
#   The find_target routine (arch/x86/ia32/ptrace32.c) doesn't deal with a
#   NULL return from  find_task_by_pid properly - if NULL is returned
#   put_task_struct() is still called.
# 
# arch/x86_64/ia32/ptrace32.c
#   2005/01/06 12:29:52-02:00 mbellon@mvista.com +3 -3
#   32 bit ltrace oops when tracing 64 bit executable [X86_64]
# 
