diff -urN kernel-source-2.4.27-2.4.27.orig/arch/s390/kernel/traps.c kernel-source-2.4.27-2.4.27/arch/s390/kernel/traps.c
--- kernel-source-2.4.27-2.4.27.orig/arch/s390/kernel/traps.c	2002-11-28 16:53:11.000000000 -0700
+++ kernel-source-2.4.27-2.4.27/arch/s390/kernel/traps.c	2006-01-29 23:12:01.000000000 -0700
@@ -619,6 +619,20 @@
 	}
 }
 
+asmlinkage void space_switch_exception(struct pt_regs * regs, long int_code)
+{
+        siginfo_t info;
+
+	/* Set user psw back to home space mode. */
+	if (regs->psw.mask & PSW_MASK_PSTATE)
+		regs->psw.mask |= PSW_ASC_HOME;
+	/* Send SIGILL. */
+        info.si_signo = SIGILL;
+        info.si_errno = 0;
+        info.si_code = ILL_PRVOPC;
+        info.si_addr = get_check_address(regs);
+        do_trap(int_code, SIGILL, "space switch event", regs, &info);
+}
 
 
 /* init is done in lowcore.S and head.S */
@@ -643,7 +657,7 @@
         pgm_check_table[0x13] = &special_op_exception;
  	pgm_check_table[0x14] = &do_pseudo_page_fault;
         pgm_check_table[0x15] = &operand_exception;
-        pgm_check_table[0x1C] = &privileged_op;
+        pgm_check_table[0x1C] = &space_switch_exception;
 #ifdef CONFIG_PFAULT
 	if (MACHINE_IS_VM) {
 		/* request the 0x2603 external interrupt */
