--- kernel-source-2.4.27-2.4.27/ChangeLog	1970-01-01 01:00:00.000000000 +0100
+++ kernel-source-2.4.27-2.4.27/ChangeLog	2004-08-07 16:28:26.000000000 -0700
@@ -0,0 +1,1261 @@
+final:
+
+- 2.4.27-rc6 was released as 2.4.27 with no changes.
+
+Here is a list of the most important security issues fixed by this release:
+
+CAN-2004-0495 (Al Viro sparse fixes)
+CAN-2004-0497 (users could modify group ID of arbitrary files on the system)
+CAN-2004-0535 (e1000 minor info leak)
+CAN-2004-0685 (backported Conectiva usb sparse fixes)
+CAN-2004-0415 (file offset pointer handling race)
+CAN-2004-0565 (information leak ia64)
+
+
+Summary of changes from v2.4.27-rc5 to v2.4.27-rc6
+============================================
+
+Geert Uytterhoeven:
+  o Fix net/atm/br2684.c file offset patch
+
+Marcelo Tosatti:
+  o Changed EXTRAVERSION to -rc6
+
+Mikael Pettersson:
+  o drivers/macintosh/nvram.c typo
+
+
+Summary of changes from v2.4.27-rc4 to v2.4.27-rc5
+============================================
+
+Adrian Bunk:
+  o [NET]: Update CONFIG_NET_SCH_NETEM Configure.help entry
+
+Chris Wright:
+  o ethtool_get_regs copy right number of bytes to user
+
+Eric Sandeen:
+  o [XFS] Don't lock down user pages when doing direct IO; this can lead to trouble (double-locking zero page, etc).
+
+Harald Welte:
+  o [NETFILTER]: ipt_ULOG fix for packet delay
+  o [NETFILTER]: Fix broken debug assertion
+
+Marcel Holtmann:
+  o [Bluetooth] Fix kernel panic when device config fails
+  o [Bluetooth] Replace BCSP retransmitting message with BT_DBG
+  o [Bluetooth] Fix resetting to default filters
+  o [Bluetooth] Send HCI_Reset for ISSC USB dongles
+
+Marcelo Tosatti:
+  o Revert DVD-RW write support for now: axboe@suse.de|ChangeSet|20040606235035|46544
+  o Cset exclude: axboe@suse.de|ChangeSet|20040607195639|57919
+  o Remove mm/page_alloc.c debugging
+  o Al Viro and others: Fix file offset handling races in several drivers
+  o Changed EXTRAVERSION to -rc5
+  o update-i386-defconfig.patch
+
+Nathan Scott:
+  o [XFS] Fix data loss problem - we no longer update i_size anywhere without holding i_sem for 2.4 as well.
+  o [XFS] Fix diotest4 test case issues with direct reads in XFS
+
+Ralf Bächle:
+  o Fix non-use of HZ in 6pack.c
+
+Stephen Hemminger:
+  o bridge fix
+  o [TCP]: Bic tcp congestion calculation timestamp
+  o [PKT_SCHED]: netem limit not returned correctly
+
+
+Summary of changes from v2.4.27-rc3 to v2.4.27-rc4
+============================================
+
+<ajgrothe:yahoo.com>:
+  o [CRYPTO]: Add TEA and XTEA algorithms
+
+<jaap.keuter:xs4all.nl>:
+  o [IPV4]: Calculate default broadcast even when using SIOCSIGNETMASK
+
+<mjc:redhat.com>:
+  o USB: more sparse fixes
+
+<tomd:csds.uidaho.edu>:
+  o [CRYPTO]: Set CRYPTO_TFM_RES_BAD_KEY_LEN in twofish
+
+Adrian Bunk:
+  o [IPV4]: Remove no longer available URL
+  o cmpci oops on rmmod + fix
+
+Anton Blanchard:
+  o [TG3]: Missing rmb() in rx processing
+
+Arun Sharma:
+  o ia64: tighten FPH state context switch check
+
+Dave Kleikamp:
+  o JFS: Error path released metadata page it shouldn't have
+  o JFS: Updated field isn't always written to disk during truncate
+  o JFS: Protect active_ag with a spinlock
+  o JFS: prevent concurrent calls to txCommit on the imap inode
+  o JFS: Check for dmap corruption before using leafidx
+  o JFS: jfs_dmap build fix
+
+David S. Miller:
+  o [TG3]: Always do 4gb tx dma test, and fix the test
+  o [TG3]: Fibre PHY fixes from Sun
+  o [TG3]: Update driver version and reldate
+  o [TG3]: Delay both before and after PCI cfg space readback after reset
+  o [TG3]: Bump driver version and reldate
+  o [TG3]: Update reldate to match 2.6.x sources
+  o [IPV4]: Make raw sockets behave like udp wrt. MSG_TRUNC
+  o [ATM]: Update Marko Kiiskila's email address
+  o [PKT_SCHED]: sch_netem.c needs linux/init.h
+  o [CRYPTO]: No MODULE_ALIAS in 2.4.x
+
+Geert Uytterhoeven:
+  o M68k ifpsp060
+  o M68k 68060 errata I14
+  o M68k Maintainership
+
+Herbert Xu:
+  o [CRYPTO]: Fix stack overrun in crypt()
+
+Jochen Hein:
+  o Update Jochen CREDITS entry
+
+Karsten Keil:
+  o I4L: Fix IRQ-sharing lockup in nj_s
+
+Marcel Holtmann:
+  o [Bluetooth] Respond to L2CAP info requests
+  o [Bluetooth] Add support for another ALPS module
+  o [Bluetooth] Use a signed integer for the RSSI value
+
+Marcelo Tosatti:
+  o USB: fix endless resubmit in auerswald (Wolfgang Mues)
+  o Changed EXTRAVERSION to -rc4
+
+Masanari Iida:
+  o Fix harmless typo in drivers/char/sysrq.c
+
+Mikael Pettersson:
+  o cardbus.c pointer truncation bug on 64-bitters
+
+Patrick McHardy:
+  o [IPV4/IPV6]: Add myself to MAINTAINERS
+
+Pete Zaitcev:
+  o USB: update unusual_devs.h
+  o USB: GET_ID from nonzero interface (errandir_news@mph.eclipse.co.uk)
+  o USB: add free_len=0 initialization to ipaq.c (Ganesh Varadaraja)
+  o USB: correct dbg() arguments in pl2303 (Phil Dibowitz)
+  o USB: missing rcomplete=0 in printer.c (David Woodhouse)
+
+Petr Vandrovec:
+  o [VLAN]: Do not access released memory
+
+Samuel Thibault:
+  o [UDP]: Return true length if user specifies MSG_TRUNC
+
+Stephen Hemminger:
+  o [PKT_SCHED]: Update to network emulation QOS scheduler
+  o [PKT_SCHED]: One small netem fixes
+  o [BRIDGE]: Fix assertion failure in 2.4.27-rc3
+  o [PKT_SCHED]: netem update for 2.4
+
+Stéphane Eranian:
+  o ia64: fix info in /proc/pal/*/bus_info
+  o ia64: fix perfmon buffer init
+
+
+Summary of changes from v2.4.27-rc2 to v2.4.27-rc3
+============================================
+
+Adrian Bunk:
+  o add missing USB Gadget Configure.help entries
+
+Christoph Hellwig:
+  o [NETLINK]: Fix NLMSG_OK/RTA_OK length checking
+
+David S. Miller:
+  o [NET]: Fix SO_{RCV,SND}TIMEO getsockopt handling
+
+Jan-Benedict Glaw:
+  o [NETFILTER]: ip_fw_compat_masq.c needs net/ip.h
+
+Marcelo Tosatti:
+  o Thomas Biege: Fix missing DAC check on sys_chown
+  o Changed EXTRAVERSION to -rc3
+  o Add missing bracket to inode_change_ok() fix
+
+Stephen Hemminger:
+  o [BRIDGE]: Bridge STP message age
+
+Tom Rini:
+  o Li Yang: PQII FCC Ethernet driver: transmit buffer leak
+
+
+Summary of changes from v2.4.27-rc1 to v2.4.27-rc2
+============================================
+
+<glen:imodulo.com>:
+  o Fix comment typo in nForce2 C1halt fixup
+
+<skraw:ithnet.com>:
+  o orinoco_pci.c new device 0x3872
+
+Adrian Bunk:
+  o add ATM_FORE200E_USE_TASKLET Configure.help entry
+  o add missing E1000_NAPI Configure.help text
+  o add SATA Configure.help texts
+
+Christoph Hellwig:
+  o [XFS] Don't leak locked pages on readahead failure
+
+Dave Kleikamp:
+  o jfs warning fix
+
+David Brownell:
+  o usb gadget API updates (1/5)
+  o usb gadget zero updates (2/5)
+  o usb file storage gadget updates (3/5)
+  o usb ethernet+RNDIS gadget updates (4/5)
+  o usb gadget build updates (5/5)
+
+David S. Miller:
+  o [PKT_SCHED]: Do not check netif_queue_stopped() in dequeue ops, races with driver
+  o [NET]: Check __dev_get_by_name() return value in eql.c
+
+David Stevens:
+  o [IPV6]: Handle user asking for any device in mcast calls
+
+Dean Roehrich:
+  o [XFS] Change things to use new version of xfs_dm_init/xfs_dm_exit
+  o [XFS] Fix non-dmapi build
+
+Eric Sandeen:
+  o [XFS] Fix overflow in mapping test at offsets of 2^63-1 bytes
+
+Geert Uytterhoeven:
+  o affs remount
+  o Mac Sonic Ethernet fixes
+  o Mac IOP: Fix bug found by Opera
+
+Harald Welte:
+  o [NETFILTER]: Fix non-existant config option
+  o [NETFILTER]: Complain when broken ipt_owner options are used on SMP
+  o [NETFILTER]: Change permissions of /proc/net/ip_conntrack to 0440
+  o [NETFILTER]: skip internal targets in iptables proc listing
+  o [NETFILTER]: Fix inverted matching in ipt_helper
+  o [NETFILTER]: 'any' matching in ipt_helper
+  o [NETFILTER]: Don't reroute on nfmark change in mangle table when routing by nfmark is not enabled
+  o [NETFILTER]: Fix truncated fragment check in ipt_unclean
+  o [NETFILTER]: Remove broken check for cleared IP_DF flag on fragments in ipt_unclean
+  o [NETFILTER]: Add new function 'nf_reset' to reset netfilter related skb-fields
+
+Hideaki Yoshifuji:
+  o [IPV6]: Fix autoconf description in ip-sysctl.txt
+
+Jeb J. Cramer:
+  o e1000 management reset fix
+
+Jeff Garzik:
+  o [netdrvr] fix ethtool_ops design bug, sync with 2.6.x ethtool_ops code
+  o Rename 'carmel' block driver to 'sx8'
+
+Manfred Spraul:
+  o yenta: Add override_bios flag to ignore BIOS resource allocation
+
+Marcel Holtmann:
+  o [Bluetooth] Kill duplicate includes
+  o [Bluetooth] Update help entries
+  o [Bluetooth] Allocate protocol number for HIDP support
+  o [Bluetooth] Add quirk for broken RTX Telecom based dongles
+  o [Bluetooth] Fix connection creation error handling
+
+Marcelo Tosatti:
+  o Geert Uytterhoeven: Fix warnings and cleanup debug_page() addded in -rc1
+  o Al Viro sparse fixes: decnet user pointer dereference
+  o Al Viro sparse fixes: mpu401 user pointer dereference
+  o Al Viro sparse fixes: msnd user pointer dereference & assorted fixes
+  o Al Viro sparse fixes: pss user pointer dereference
+  o Al Viro sparse fixes: aironet
+  o Al Viro sparse fixes: asus_acpi user pointer dereference
+  o Changed EXTRAVERSION to -rc2
+
+Matt Domsch:
+  o edd.c display %u, remove REPORT_URL
+
+Mikael Pettersson:
+  o i386 and x86_64 ACPI mpparse timer bug
+
+Nathan Scott:
+  o [XFS] No longer hold the BKL for the entire ioctl operation, its not needed here.
+  o [XFS] Remove a couple of redundant NULL parent inode pointer checks
+  o [XFS] Fix xfs_lowbit64, it mishandled zero in the high bits
+  o [XFS] sparse: fix uses of macros before their definitions, etc
+  o [XFS] Ensure buffers that map to unwritten extents are only submitted when properly setup.
+  o [XFS] Sanitise the ACL initialisation macros
+  o [XFS] Remove unused MAC macros, never needed on Linux
+  o [XFS] Remove the one remaining, broken use of XFS_WRITEIO_LOG and sanitize direct IO map blocks call.
+  o [XFS] Fix flags argument to xfs_incore call on attr removal
+  o [XFS] Fix a race condition in the undo-delayed-write buffer routine
+  o [XFS] Fix up memory allocators to be more resilient
+  o [XFS] Fix up error handling on inode shrink register
+  o [XFS] Fix up memory reclaim interfaces for 2.4 kernels
+
+Pete Zaitcev:
+  o Lonnie Mendez: Remove USB extra #include
+
+Russell Cattelan:
+  o [XFS] Fix for NFS+XFS data corruption problem
+
+Scott Feldman:
+  o e1000: fix napi crash on ifdown during traffic
+
+Stephen Hemminger:
+  o [PKT_SCHED]: Delay scheduler enqueue always succeeds
+  o [PKT_SCHED]: Delay scheduler should retry if requeue fails
+  o [PKT_SCHED]: Add loss option to network delay scheduler
+  o [BRIDGE]: Backport of API checking
+  o [BRIDGE]: Elimintate br_ioctl_mutex
+
+
+Summary of changes from v2.4.27-pre6 to v2.4.27-rc1
+============================================
+
+<andreas:xss.co.at>:
+  o fix hotplug Config.in xconfig breakage
+
+Andi Kleen:
+  o Undo mistaken hunks in previous x86-64 MCE change
+  o Add missing include to x86-64 bluesmoke.c
+
+Chris Wedgwood:
+  o stat nlink resolution fix
+
+David S. Miller:
+  o Cset exclude: kaber@trash.net|ChangeSet|20040529193918|20643
+
+David Stevens:
+  o [IPV4]: Fix interface selection in multicast sockops
+
+Eyal Lebedinsky:
+  o Fix USB visor.c compilation error
+
+Hideaki Yoshifuji:
+  o [IPV6]: UDPv6 checksum
+  o [IPV6]: UDPv6: Use udpv6_queue_rcv_skb()
+  o [IPV6]: Missing include in ip6_tables.c
+
+Karol Kozimor:
+  o acpi4asus trivial sync with 2.6 (Karol 'sziwan' Kozimor)
+
+Len Brown:
+  o [ACPI] PCI bus numbering workaround for ServerWorks from David Shaohua Li http://bugzilla.kernel.org/show_bug.cgi?id=1662
+  o [ACPI] fix passive cooling mode indicator (Luming Yu) http://bugzilla.kernel.org/show_bug.cgi?id=1770
+  o [ACPI] avoid spurious interrupts on VIA http://bugzilla.kernel.org/show_bug.cgi?id=2243
+  o [ACPI] fix 2.4.27-pre3 IRQ override regression due to dynamically allocated mp_irqs[].
+  o [ACPI] handle SCI override to nth IOAPIC http://bugzilla.kernel.org/show_bug.cgi?id=2835
+
+Marcelo Tosatti:
+  o Add missing struct definition of rwsem race fixes
+  o Changed EXTRAVERSION to -rc1
+  o journal_try_to_free_buffers(): Add debug print in case of bh list corruption
+
+Mikulas Patocka:
+  o HPFS fixes
+
+Pete Zaitcev:
+  o USB: Fix jumpshot's capacity
+
+
+Summary of changes from v2.4.27-pre5 to v2.4.27-pre6
+============================================
+
+<andrej.filipcic:ijs.si>:
+  o USB: pl2303 & input overruns
+
+<chaapala:cisco.com>:
+  o [CRYPTO]: Fix digest.c kmapping sg entries > page in length
+
+<hadi:zynx.com>:
+  o [NETFILTER]: Small interface cleanup for {ipt,ip6t,arpt}_find_target
+
+<jan:ccsinfo.com>:
+  o USB ftdi device ids for 2.4
+
+<jhh:lucent.com>:
+  o [SCTP] Fix to not setup a new association if the endpoint is in SHUTDOWN_ACK_SENT state and recognizes that the peer has restarted. 
+  o [SCTP] Fix to not start a new association on a 1-many style sendmsg() with MSG_EOF/MSG_ABORT flag and no data.
+
+<juhl-lkml:dif.dk>:
+  o [NET]: Remove junk from packet_mmap.txt
+
+<kumar.gala:freescale.com>:
+  o Simple build fix for PPC 826x
+
+<martin.lubich:gmx.at>:
+  o USB: add Clie TH55 Support in visor kernel module
+
+<nickpiggin:yahoo.com.au>:
+  o rwsem race fixes backported from 2.6
+
+<raven:themaw.net>:
+  o minor autofs4 fs/namei.c fix, URL update
+
+<shurick:sectorb.msk.ru>:
+  o i2c-matroxfb and i2c initialization order
+
+Andi Kleen:
+  o Fix argument parsing in x86-64 machine check handler
+  o Fix LDT/TSS limit on x86-64
+  o Fix compilation without CONFIG_SWIOTLB on x86-64
+  o Fix boot loader warnings on x86-64
+  o Add missing memory clobber to i386
+  o Fix K8 machine check decoding
+  o Fix FPU delayed exceptions on x86-64 too
+
+Andreas Dilger:
+  o [IPV4]: Fix bug in arp_tables.c fix
+
+Bartlomiej Zolnierkiewicz:
+  o clarify help entries for Promise IDE drivers a bit
+  o ide: add new nForce IDE/SATA device IDs to amd74xx.c
+
+Ben Collins:
+  o video1394: Bugfix for low res format7 images
+  o sbp2: Default sbp2_serialize_io to 1
+  o pcilynx: Disable LCtrl bit with IRM contender
+
+Dave Kleikamp:
+  o JFS: Handle out of space errors more gracefully
+  o JFS: Better RAS when btstack is overrun
+  o JFS: Don't allow reading beyond the inode map's EOF
+  o JFS: Fix compilation error
+
+David Brownell:
+  o EHCI fixes (byteswap, BIOS)
+  o EHCI fixes (byteswap, BIOS)
+
+David S. Miller:
+  o [TG3]: Use HOST TXDs always
+  o [IPV4]: Fix unaligned accesses in arp_tables.c
+  o [TG3]: Chip support update and a power-save bug fix
+  o [TG3]: Update driver version and reldate
+  o [SPARC]: Report si_addr in SIGINFO more accurately
+
+Greg Kroah-Hartman:
+  o USB: fix empty write issue in pl2303 driver
+
+Harald Welte:
+  o [NETFILTER]: Don't assign new helper after NAT when there are already expectations present
+
+Herbert Xu:
+  o Fix /proc/ide/hpt366 crash
+
+Hideaki Yoshifuji:
+  o [UDPv4]: Pass correct socket to ip_mc_sf_allow
+
+Ivan Kokshaysky:
+  o Alpha: don't put IDE disks in standby mode on halt
+  o Alpha: fix PCI bridge swizzle on takara and eiger
+
+Jamal Hadi Salim:
+  o [NET]: Add ARPHRD_NONE and use it in tun driver
+
+Jens Axboe:
+  o DVD-RW write support
+  o cdrom hardware defect mgt header length
+
+Marcelo Tosatti:
+  o Cset exclude: marcelo@logos.cnet|ChangeSet|20040606231521|43962
+  o Thomas Winischhofer: sisfb update
+  o Alexander Nyberg/Andi/Sergey: Fix x86 "clear_cpu()" macro
+  o Changed EXTRAVERSION to -pre6
+  o Michael Reinelt: Add support for NetMos 9835 serial cards
+  o John Carlson: Remove bogus ";" from USB gadgets's usb_descriptor_fillbuf
+
+Mike Miller:
+  o cciss update
+
+Neil Brown:
+  o Fix hard-to-hit BUG in raid5 resync code
+
+Nuno Monteiro:
+  o Fix rwsem-fix typo
+  o Complete rwsem typo fix
+
+Pete Zaitcev:
+  o Fix USB serial race
+  o USB: Update mct_u232
+  o USB: unusual_dev.h add Sony Handycam HC-85
+
+Sridhar Samudrala:
+  o [SCTP] Fix the use of cached non-zero vtag in an INIT that is resent after a stale cookie error.
+  o [SCTP] Fix missing VTAG validation on certain incoming packets
+  o [SCTP] Fix to wakeup blocking connect() after max INIT retries failed
+  o [SCTP] Fix poll() on a 1-1 style socket so that it returns when the association is aborted by peer.
+
+
+Summary of changes from v2.4.27-pre4 to v2.4.27-pre5
+============================================
+
+<chrisg:etnus.com>:
+  o x86-64 only: ia32entry.S reg changes dropped during debugging
+
+<kevin.curtis:farsite.co.uk>:
+  o [netdrvr wan] farsync update
+
+<pelle:dsv.su.se>:
+  o tulip PCI ID for D-Link DFE-680TXD
+
+<vda:port.imtp.ilyichevsk.odessa.ua>:
+  o fealnx #0: replace dev->base_addr with ioaddr; sync with 2.6
+  o fealnx #1: replace magic constants with enums
+  o fealnx #2: add 'static'; fix wrapped comment
+  o fealnx #3: fix pointer substraction bug
+  o fealnx #4: stop doing stop_nic_rx/writel(np->crvalue) in rest_rx_descriptors()
+  o fealnx #5: introduce stop_nic_rxtx(), use it where makes sense
+  o fealnx #6: Francois' fixes for low memory handling; remove free_one_rx_descriptor (not used anymore)
+  o fealnx #7: Garzik fix (IIRC): add locking to tx_timeout
+  o fealnx #8
+  o fealnx #9
+  o fealnx #10
+  o fealnx #11
+
+Andrew Morton:
+  o 8139too not running s3 suspend/resume pci fix
+
+Arjan van de Ven:
+  o small change for scsi 2.6 header compatibility
+  o [libata] Use standard headers from include/scsi, not drivers/scsi
+
+Daniele Venzano:
+  o [netdrvr sis900] fix ISA bridge detection
+  o [netdrvr sis900] cosmetic header cleanups
+  o [netdrvr sis900] fix missing netif_device_detach() in suspend
+
+David S. Miller:
+  o [TCP]: Fix build in 2.4.x with SCTP disabled
+  o [PKT_SCHED]: Missing rta_len init in sch_delay
+  o [SPARC64]: gcc-3.4.x build fixes
+
+Don Fry:
+  o 2.4.27-pre3 pcnet32 add static to two routines
+  o 2.4.27-pre3 pcnet32 avoid hard hang with some chip variants
+  o 2.4.27-pre3 pcnet32 correct 79C976 variant string
+  o 2.4.27-pre3 pcnet32 fix boundary comparison bug
+  o 2.4.27-pre3 pcnet32 remove timer and complexity
+  o pcnet32: limit frames received during interrupt
+  o pcnet32: fix bogus carrier errors with 79c973
+  o pcnet32: correct printk for big-endian arch
+  o pcnet32: avoid timeout with tcpdump
+  o pcnet32: fix for patch 8 le16_to_cpu
+
+François Romieu:
+  o [netdrvr r8169] update to 2.6.x version; many fixes and changes
+
+Ganesh Venkatesan:
+  o e1000 1/7: Clear auto-mdix mode when forcing link to
+  o e1000 2/7: Workaround for link LED staying ON even when
+  o e1000 3/7: Determine Link Status correctly while using
+  o e1000 4/7: Rewrite logic to estimate # of tx descriptors
+  o e1000 6/7: ethtool_ops support
+  o e1000 7/7: Error Logging support (enabled/disabled via
+  o e100 1/1: Update to sync up version numbers
+
+Jeff Garzik:
+  o [sound i810] pci id cleanups
+  o [libata] add new ->bmdma_setup hook
+  o [libata] use new ->bmdma_{start,setup} method to properly support ATAPI
+  o [libata] more ATAPI work - translate SCSI CDB to ATA PACKET
+  o [libata] random minor bug fixes
+  o [libata] kill ATA_QCFLAG_POLL flag
+  o [libata] internal cleanups
+  o [libata] minor stuff
+  o [libata] handle non-data ATAPI commands via interrupt
+  o [libata] DMADIR support
+  o [libata] remove redundant use of ATA_QCFLAG_SG in ATAPI packet translation
+  o [libata] SCSI->ATA simulator hacking: INQUIRY command
+  o [libata] comments and constants
+  o [libata] scsi simulator improvements: MODE SENSE, SEEK(6,10), REZERO_UNIT
+  o [libata] replace ATA_QCFLAG_ATAPI with inline helper
+  o [libata] polish DocBook docs a bit
+  o [netdrvr sis900] sync with 2.6.x
+  o [netdrvr e1000] use generic ethtool_ops provided in net/core/ethtool.c
+  o [libata promise] revert broken taskfile delivery change
+  o [libata scsi] ack SYNCHRONIZE CACHE command
+
+Marcelo Tosatti:
+  o Changed EXTRAVERSION to -pre5
+
+Patrick McHardy:
+  o [IPV4,6]: Fix off-by-one in max protocol-type check
+
+Stefan Rompf:
+  o [netdrvr b44] always restore PCI config on resume
+
+
+
+Summary of changes from v2.4.27-pre3 to v2.4.27-pre4
+============================================
+
+<akepner:sgi.com>:
+  o [TG3]: Fix ethtool -S
+  o [TG3]: Make sure RX/TX flow control settings actually get set
+
+<felixb:sgi.com>:
+  o [XFS] Remove speculative preallocation from linvfs_get_block_core
+
+<fsgqa:sgi.com>:
+  o [XFS] Remove the 128K limitation on pagebuf_get_no_daddr() and allow the kmem_alloc() to fail.
+
+<gnb:melbourne.sgi.com>:
+  o [TG3]: Count rx_discards in rx_errors
+  o [TG3]: Add more ethtool -S stats
+
+<joshk:triplehelix.org>:
+  o [SPARC64]: Pull in drivers/i2c/Config.in
+  o [SPARC64]: Use $(CC) in check for egcs/gcc3
+  o [SPARC64]: Backport some lvalue warning fixes from 2.6.x
+  o [SPARC32]: Use $(LD) -V for NEW_GAS check
+
+<jpk:sgi.com>:
+  o [XFS] xfs_iomap_write_delay() was doing speculative allocations without checking if there were any real blocks already in the speculative allocation area. This could result in an allocation that overlaps pre-allocated space. This would result in an ASSERT failure in debug kernels, or invalid output from xfs_bmap.
+  o [XFS] Add support for allocating additional file space in stripe width sized chunks. A new fstab/mount option, "swalloc" has been defined. If specified when mounting a striped file system, allocation requests will be rounded up to a stripe width if the file size is >= stripe width, and the data is being appended to eof. The 'swalloc' option is "off" by default.
+
+<linux-kernel:vortech.net>:
+  o [VLAN]: Use KERN_INFO for VLAN_INF
+
+<mhuth:mvista.com>:
+  o [IPV6]: Fix sock identity checking bug in tcp_ipv6_check_established
+
+<michael.waychison:sun.com>:
+  o cramfs use pagecache
+
+<tony.cureington:hp.com>:
+  o [TG3]: Add eeprom dump support
+
+<zli4:cs.uiuc.edu>:
+  o [SPARC]: Fix prom_prom_taken[].theres_more setting
+
+Adrian Bunk:
+  o [ATALK]: Fix modular build
+  o [NET]: Missing MODULE_LICENSE in p8022 and psnap
+  o SCSI ips compile error
+
+Andi Kleen:
+  o Fix pageattr cache flushing on P4 (thanks to Terence Ripperda)
+
+Andrew Morton:
+  o use-before-uninitialized value in ext3(2)_find_ goal
+
+Christoph Hellwig:
+  o [XFS] fix direct user memory dereference in bulkstat
+  o [XFS] Use macros instead of inlines for spinlock wrappers to aid debugging.
+
+Christophe Saout:
+  o [CRYPTO]: Fix two scatterwalk problems
+
+Dave Kleikamp:
+  o JFS: error in __get_metapage caused by invalid size from ea_get
+  o JFS: Don't return -EPERM for system xattrs
+  o JFS: Make sizes of tid_t & lid_t consistent
+  o JFS: [CHECKER] Memory leak on commonly executed path
+  o JFS: [CHECKER] Dereference of NULL pointer if alloc_metapage fails
+  o JFS: [CHECKER] if txCommit fails, don't call d_instantiate
+  o JFS: fix hang in __get_metapage
+
+David S. Miller:
+  o [TG3]: Add 572x/575x PCI IDs
+  o [TG3]: Add 5750 chip and PHY IDs
+  o [TG3]: Prepare for 5750 support plus minor fixes
+  o [TIGON3]: Detect and record PCI Express
+  o [TG3]: PCI Express 5750_A0 chips need 5701_REG_WRITE_BUG treatment
+  o [TG3]: Fix chiprev test in previous change
+  o [TG3]: Do not set CLOCK_CTRL_DELAY_PCI_GRANT on PCI Express
+  o [TG3]: Double delay after writing MAC_MI_MODE reg
+  o [TG3]: Correct RDMAC/WDMAC mode settings on 5705/5750
+  o [TG3]: Do not write stats coalescing ticks reg on 5705/5750
+  o [TG3]: Update to 5788 capable 5705 TSO firmware, version 1.2.0
+  o [TG3]: Update to non-5705 TSO firmware version 1.6.0
+  o [TG3]: If asked to load TSO firmware on 5750, just return success
+  o [TG3]: Add 5750 NVRAM programming plus 5704 MAC offset bug fix
+  o [TG3]: Update LED programming to support 5750
+  o [TG3]: Updated ASF handling for 5750
+  o [TG3]: Include mss in every txd, not just the first, on 5750
+  o [TG3]: On 5750 with TSO, need to set some special reg bits
+  o [TG3]: Full chip reset tweaks for 5750
+  o [TG3]: More 5750 chip reset tweaks
+  o [TG3]: Do not enable slow clocks on 5750 with ASF
+  o [TG3]: Rewrite dma_rwctrl settings to handle PCIX/PCIE
+  o [TG3]: Add 572x/575x PCI IDs to driver table, update vers/reldate
+  o [SPARC64]: Update defconfig
+  o [TG3]: Update driver version and reldate
+  o [NET]: Fix common typo, using NODEV when we mean ENODEV
+  o [TG3]: Fix phantom spaces added to pci.ids file
+  o [TCP]: Kill distance enforcement between tcp_mem[] elements
+  o [TCP]: Abstract out all settings of tcp_opt->ca_state into a function
+  o [TCP]: Backport Vegas support from 2.6.x
+  o [TCP]: Backport BIC TCP from 2.6.x
+  o [TCP]: Add tcp_default_win_scale sysctl
+  o [TCP]: Add receiver side RTT estimation
+  o [TCP]: Grow socket receive buffer based upon estimated sender window
+  o [TCP]: More sysctl tweakings for rcvbuf stuff
+
+Dean Roehrich:
+  o [XFS] Dmapi preunmount event references null pointer
+
+Dely Sy:
+  o Documentation for new PCI hotplug drivers
+
+Dmitry Torokhov:
+  o [NET_SCHED]: Do not oops when user tries to attach a filter to a TBF qdisc
+
+Eric Dean Moore:
+  o MPT Fusion driver 2.05.16 update
+
+Eric Sandeen:
+  o [XFS] guard against unused var in new mutex_spinunlock #define
+
+François Romieu:
+  o CPUID Pentium-4E update and missing new line
+
+Hideaki Yoshifuji:
+  o [IPV6] ensure to evaluate the checksum for sockets with the IPV6_CHECKSUM option
+  o [NET]: Prevent future missed updates of FOO_MAX macros
+
+Hugh Dickins:
+  o tmpfs surplus page miscounted
+
+Jakub Bogusz:
+  o missing include in drivers/sound/kahlua.c
+
+Jeff Garzik:
+  o [netdrvr tg3] netdev_priv
+  o [netdrvr tg3] sync with 2.6.x
+
+Len Brown:
+  o [ACPI] delete IOAPIC-disable workaround on x86_64/VIA BTW. looks like 2.6 has an IOMMU disable workaround here that may be needed or VIA in 2.4.
+  o [ACPI] revert button module unload fix (OSDL 2281) Cset exclude: len.brown@intel.com|ChangeSet|20040504154434|56458 Cset exclude: len.brown@intel.com|ChangeSet|20040428081912|57065 Cset exclude: len.brown@intel.com|ChangeSet|20040428054017|55837
+  o [ACPI] remove /proc files before unloading modules from Sau Dan Lee, Zhenyu Wang http://bugzilla.kernel.org/show_bug.cgi?id=2705
+  o [ACPI] x86_64 + ACPI + IOAPIC + PCI Interrupt Link -> IRQ 0 from Andy Currid
+
+Marcelo Tosatti:
+  o ext3_read_super: remove [un]lock_super  (Andrew Morton)
+  o Changed EXTRAVERSION to -pre4
+
+Nathan Scott:
+  o Fix an incorrect email address in XFS maintainers section
+  o [XFS] Remove unused transaction pointer from bulkstat
+  o [XFS] Bump the kmalloc/vmalloc cutoff up to 128k
+  o [XFS] Make uses of extended inode flags consistent, remove duplicated code
+  o [XFS] Fix some compiler warnings, mark cmn_err as printflike
+  o [XFS] Fixup a couple of incorrect xfs_trans_commit calls (bad flags/casts).
+  o [XFS] Export/import tunable time intervals as centisecs not jiffies
+  o [XFS] Switch all XFSDEBUG to DEBUG
+  o [XFS] Fix a use-after-free during transaction commit when the log is in error state.
+  o [XFS] Use set_current_state instead of direct current->state assignment
+  o [XFS] Remove xfs_swappable code, its not useful on Linux
+  o [XFS] Remove no-longer-used variable in log write code, and a dated comment.
+  o [XFS] Remove unused xfs_trans_bhold_until_committed and related macros
+  o [XFS] Rename a subdirectory to make life easier for people (esp
+
+Oleg Drokin:
+  o Fix possible memleaks in VIA IrDA driver
+
+Patrick McHardy:
+  o [IPV4]: Fix skb leak in igmpv3_newpack
+
+Paul Mackerras:
+  o Fix incorrect PT_FPSCR definition
+
+Sridhar Samudrala:
+  o [SCTP] Fix memset() parameter ordering
+  o [SCTP] Fix accessing Gap Ack blocks array with a -ve index in sctp_outq_sack()
+  o [SCTP]: Fix multihomed connection failures on 64-bit systems
+
+Stephen Hemminger:
+  o [BRIDGE]: Update bridge.txt
+  o [TCP]: Add sysctl to turn off metrics caching
+  o [TCP]: Add vegas sysctl docs
+
+Timothy Shimmin:
+  o [XFS] Change xfs_contig_bits to work on 32/64 and both endian styles
+
+Trond Myklebust:
+  o NFS client: Fix sillydelete()
+
+
+
+Summary of changes from v2.4.27-pre2 to v2.4.27-pre3
+============================================
+
+<achurch:achurch.org>:
+  o ext2fs sb= mount option fix
+
+<chad.dupuis:hp.com>:
+  o Fix HP HSG80 storage array entry on SCSI device list
+
+<gtw:cs.bu.edu>:
+  o [sound i810] silently ignore invalid PCM_ENABLE_xxx bits from userland
+
+<jparmele:wildbear.com>:
+  o ad1848: Fix crystal CS4235 mixer setup
+
+<lkml:lievin.net>:
+  o tipar char driver: wrong timeout value
+
+<m.c.p:kernel.linux-systeme.com>:
+  o copy WARN_ON() definition from 2.6
+
+<ouellettes:videotron.ca>:
+  o Remove extra semicolumn in arch/i386/mm/fault.c
+
+<ross:datscreative.com.au>:
+  o nVidia supplied nForce2 workaround
+
+Andrew Morton:
+  o sata_sx4.c warning fix
+
+Arun Sharma:
+  o ia64: make ia32 core dumps work
+
+Bjorn Helgaas:
+  o Cset exclude: arun.sharma@intel.com[helgaas]|ChangeSet|20040405163028|42090
+  o ia64: Fix ia32 core dumps
+  o ia64: Update defconfigs
+  o ia64: Update defconfig to current generic config
+  o ia64: Allow IO port space without EFI RT attribute
+  o ia64: Define acpi_pci_disabled for recent ACPI update
+
+Dave Kleikamp:
+  o JFS: [CHECKER] Fix a possible null-pointer dereference
+  o JFS: [CHECKER] Memory leak in jfs_link
+  o JFS: [CHECKER] get rid of txAbortCommit
+  o JFS: Avoid race invalidating metadata page
+  o JFS: reduce stack usage
+  o JFS: [CHECKER] More robust error recovery in add_index
+  o JFS: module unload was not removing /proc/fs/jfs/
+
+David Mosberger:
+  o ia64: Fix typo in unwinder which could cause NULL-pointer dereferences
+
+Don Fry:
+  o pcnet32 whitespace only changes
+  o pcnet32 support for 79C976
+  o pcnet32 all printk under netif_msg
+  o pcnet32 correct name display
+  o pcnet32 add led blink capability
+  o pcnet32 transmit performance fix
+  o pcnet32 add register dump capability
+  o pcnet32 timer to free tx skbs for 79C971/972
+
+Gerd Knorr:
+  o Fix V4L miss of range check oops
+
+Herbert Xu:
+  o [sound i810] fix wait queue race in drain_dac
+  o [sound i810] fix race
+  o [sound i810] remove bogus CIV_TO_LVI
+  o [sound i810] clean up with macros
+  o [sound i810] fix partial DMA transfers
+  o [sound i810] fix playback SETTRIGGER
+  o [sound i810] fix OSS fragments
+  o [sound i810] remove divides on playback
+  o [sound i810] fix drain_dac loop when signals_allowed==0
+  o [sound i810] fix reads/writes % 4 != 0
+  o [sound i810] fix deadlock in drain_dac
+
+Jack Hammer:
+  o ServeRAID driver update to 7.00.15: sync with v2.6
+
+Jack Steiner:
+  o ia64: fix HUGETLB null pointer dereference
+
+Jeff Garzik:
+  o [netdrvr b44] sync with 2.6.x version
+  o Add dummy "__user" marker, for compat with 2.6.x
+  o Rename get_current_user tmpvar to avoid namespace clash
+  o [netdrvr] Add driver for IBM p/iSeries virtual ethernet adapters
+  o [libata sata_sis] add new PCI id
+  o [libata] Promise driver split part 1: clone to sx4
+  o [libata] Promise driver split part 2: remove SX4 code from sata_promise
+  o [libata] Promise driver split part 3: remove TX2/4 code from sata_sx4
+  o [libata] Promise driver split part 4: common header
+  o [libata] add ata_tf_{to,from}_fis helpers
+  o [libata] clean up taskfile submission to hardware
+  o [libata] remove call to WARN_ON(), 2.4 doesn't have this
+  o [libata] move ATAPI startup from katad thread to workqueue thread
+  o [libata] minor updates
+  o [libata] move PIO data xfer from katad thread to workqueue thread
+  o [libata] move probe execution from katad thread to workqueue thread
+  o [libata] move ATAPI command initiation code from libata-scsi to libata-core
+  o [libata] kill unnecessary include
+  o [libata] internal cleanup: kill ata_pio_start
+  o [libata] some work on the ATAPI path
+  o [libata] Make sure to initialize PIO data xfer state
+  o [libata] replace per-command semaphore with optional completion
+  o [libata promise] make sure our schedule_timeout(N) are never with N==0
+  o [libata] remove unused struct ata_engine
+  o [libata sata_sx4] trivial: fix filename in header
+  o [libata sata_sis] support SATA SCRs in PCI cfg space
+  o [libata] preparation for writeback caching support
+  o [libata] Maintainer annotations
+  o [sound i810] sync with 2.6.x driver
+  o [sound i810] bump driver to version 1.00
+
+Jeremy Higdon:
+  o sata_vsc initialization fix
+
+Karol Kozimor:
+  o acpi4asus 0.28 (Karol 'sziwan' Kozimor)
+
+Krzysztof Halasa:
+  o [netdrvr tulip] fix use-after-free
+
+Len Brown:
+  o [ACPI] delete unused CONFIG_ACPI_RELAXED_AML this code is included always -- or disable at boot w/ "acpi=strict
+  o [ACPI] Delete IRQ2 "cascade" in ACPI IOAPIC mode no such concept exists in ACPI, frees IRQ2 for use.
+  o [ACPI] enhance intr-src-override parsing to handle ES7000 http://bugme.osdl.org/show_bug.cgi?id=2520
+  o [ACPI] allow use of IRQ2 in ACPI/IOAPIC mode http://bugzilla.kernel.org/show_bug.cgi?id=2564
+  o [ACPI] No IRQ known... - using IRQ 255 (Bjarni Rúnar Einarsson) http://bugzilla.kernel.org/show_bug.cgi?id=2148
+  o [ACPI] battery "charged" instead of "unknown" (Luming Yu) http://bugzilla.kernel.org/show_bug.cgi?id=1863
+  o [ACPI] support button driver unload (Luming Yu) http://bugzilla.kernel.org/show_bug.cgi?id=2281
+  o [ACPI] pci-link may not always be SHARED (SuSE via Luming Yu) http://bugzilla.kernel.org/show_bug.cgi?id=2404
+  o [ACPI] pcibios_scan_root fix for IA64 (from IA64 tree via Luming Yu) http://bugzilla.kernel.org/show_bug.cgi?id=2130
+  o [ACPI] toshiba_acpi driver if acpi_disabled (David Shaohua Li) http://bugzilla.kernel.org/show_bug.cgi?id=2465
+  o [ACPI] button build fix
+  o [ACPI] export symbols to button module
+  o [ACPI] PCI Interrupt Link fixes
+  o [ACPI] rmmod ACPI modules vs /proc from Anil S Keshavamurthy and David Shaohua Li http://bugzilla.kernel.org/show_bug.cgi?id=2457
+  o [ACPI] enhance "pci=noacpi" to skip PCI probe (David Shaohua Li) add "acpi=noirq" to just disable IRQ config http://bugzilla.kernel.org/show_bug.cgi?id=1662
+  o [ACPI] parse ACPI MCFG table and set pci_mmcfg_base_addr from 2.6 via Sundar/Dely Sy
+  o [ACPI] handle _CRS outside _PRS -- even when non-zero avoid sharing IRQ12 http://bugzilla.kernel.org/show_bug.cgi?id=2665
+
+Marcelo Tosatti:
+  o Delete: drivers/char/amd7xx_tco.c
+  o Matt Domsch: Lower printk severity for PCI devices with no PCI_CACHE_LINE_SIZE implemented
+  o Changed EXTRAVERSION to -pre3
+
+Mike Miller:
+  o cciss update: support for two new controllers
+  o Fix cciss bug in proc reporting
+
+Oleg Drokin:
+  o [2.4] Make reiserfs not to crash on oom during mount
+
+Pavel Machek:
+  o Cleanups for b44
+
+Pete Zaitcev:
+  o tiglusb bug fixes (usb_clear_halt, usb_sndbulkpipe)
+  o USB: Check results of copy_to_user in hiddev
+  o USB: Fix memory leaks in speedtch
+  o USB: Fix IBM USB memory key unusual dev entry
+
+Richard Curnow:
+  o Patch [sh64]: Fix liveness dependency constraints in _syscalln()
+  o Fix Cayman PCI IRQ routing for PCI cards containing bridges
+  o Patch [sh64]: Fix approach for calibrating CPU MHz so it works even if the I-cache is disabled
+  o Patch [sh64]: Clean-up handling of the SR.WATCH bit
+  o Patch [sh64]: Remove obsolete CONFIG_UNCACHED_MEMORY_OFFSET option
+  o Patch [sh64]: clean-up to allow building on Cayman board without configuring PCI support
+  o Patch [sh64]: Handle base of RAM not being 512Mb aligned
+  o Patch [sh64]: Allow greater choice over the effective address at which the kernel runs
+
+Sridhar Samudrala:
+  o [SCTP] Fix bugs in handling overlapping INIT and peer restart over a multihomed association.
+  o [SCTP] Rename SCTP_ADDR_REACHABLE as SCTP_ADDR_AVAILABLE to be consistent with the SCTP sockets API draft.
+
+Stephen Hemminger:
+  o SCTP crc table can be static const
+
+Stéphane Eranian:
+  o ia64: switch /proc/perfmon to seq_file avoid buffer overflows
+
+Zwane Mwaikambo:
+  o remove amd7(saucy)_tco
+
+
+
+Summary of changes from v2.4.27-pre1 to v2.4.27-pre2
+============================================
+
+<aj:andaco.de>:
+  o [TG3]: Fix typo in TG3_TSO_FW_RODATA_ADDR definition
+
+<gandalf:winds.org>:
+  o Blue line in nVidia framebuffer (rivafb)
+
+<jkmaline:cc.hut.fi>:
+  o [CRYPTO]: Add suuport for keyed digests
+  o [CRYPTO]: Add Michael MIC algorithm
+
+<jpk:sgi.com>:
+  o [XFS] Correct the (file size >= stripe unit) check inside xfs_iomap_write_delay.  It was comparing the file size, in bytes, against the stripe unit size, in FSBs.
+
+<mchan:broadcom.com>:
+  o [TG3]: Jumbo frames and FTQ reset patch
+  o [TG3]: Fix jumbo frame PHY programming
+
+Andrea Arcangeli:
+  o Fix page-free-at-IRQ context BUG
+
+Bartlomiej Zolnierkiewicz:
+  o IDE PCI: fix support for PIO modes w/o IORDY flow control
+  o serverworks.c: fix DMA for OSB4 (Patrick Wildi)
+  o generic PCI IDE support for more Toshiba Piccolo chips (Daniel Drake)
+
+Chas Williams:
+  o [ATM]: get atm_guess_pdu2truesize() right
+  o [ATM]: [nicstar] using dev_alloc_skb() (reported by Johnston,
+  o [ATM]: [fore200e] 0.3e version by Christophe Lizzi (lizzi@cnam.fr)
+  o [ATM]: [fore200e] make tasklet use configurable
+
+Chris Wright:
+  o [IPV4]: Fix return value on MCAST_MSFILTER error case
+
+Christoph Hellwig:
+  o [XFS] use kmem_alloc for noaddr buffers
+  o [XFS] kill the pagebuf vs xfs_buf confusion
+  o [XFS] really kill the pagebuf vs xfs_buf confusion
+  o [XFS] clarify pagebuf page lookup logic
+  o [XFS] cleanup pagebuf flag usage and simplify pagebuf_free
+  o [XFS] close external blockdevice after final flush
+
+David S. Miller:
+  o [TG3]: Just completely delete the disabled FTQ reset code
+  o [SPARC64]: Kill cast as lvalue usage in pgd/pmd macros
+  o [SPARC64]: Fix rest of cast as lvalue usage in sparc64 code
+  o [TG3]: Kill 'force' arg to tg3_phy_reset, it is always set
+  o [TG3]: At start of tg3_phy_copper_begin, force phy out of loopback mode
+  o [TG3]: Do not allow illegal ethtool advertisement bits
+  o [SPARC64]: Export prom_palette
+  o [TG3]: Add missing 5704 BX workaround, and fix typo in autoneg fix
+  o [TG3]: Set GRC_MISC_CFG prescaler more safely
+  o [TG3]: Fix serdes cfg programming on 5704
+  o [TG3]: When link is down, set stats coalescing ticks to zero
+  o [TG3]: Wait a bit for BMSR/BMCR bits to settle in PHY setup
+  o [TG3]: Verify link advertisement correctly on 10/100 only chips
+  o [TG3]: All 5705 chips need PHY reset on link-down
+  o [TG3]: More PHY programming fixes
+  o [TG3]: Bump driver version and reldate
+  o [TG3]: Print list of important probed capabilities at driver load
+  o [TG3]: Two PHY fixes
+  o [TG3]: Kill uninitialized var warning
+  o [TG3]: Reset fixes
+  o [TG3]: Update driver version and release date
+  o [SPARC]: Use 64 for KERN_SPARC_SCONS_PWROFF to prevent 2.6.x conflicts
+  o [TG3]: Update driver version and reldate
+  o [TG3]: Undo comment typo fix, it was wrong
+  o [SPARC64]: Fix zero-extension issues wrt. {pgd,pmd}_val()
+  o [SPARC64]: Update defconfig
+
+David Stevens:
+  o [IPV4]: Fix IGMP version number and timer printing for procfs
+
+Dely Sy:
+  o SHPC and PCI Express hot-plug drivers for 2.4 kernel
+  o PCI Hotplug: SHPC & PCI-E hot-plug fixes
+
+Eric Brower:
+  o [SPARC]: Add sysctl to control serial console power-off restrictions
+  o [SPARC64]: HDIO_DRIVE_TASK is a compatible ioctl
+
+Eric Sandeen:
+  o [XFS] Use pgoff_t for page indices, and remove some other type confusion
+  o [XFS] New PFLAGS_RESTORE_FSTRANS macro to restore only FSTRANS state from saved state.
+
+Geert Uytterhoeven:
+  o [NET]: Make pktgen depend upon procfs
+  o NCR53C9x unused SCp.have_data_in
+  o M68k TLB fixes
+  o Amiga A2065 Ethernet debug
+
+Grant Grundler:
+  o [TG3]: Fix comment typo
+
+Harald Welte:
+  o [NETFILTER]: Add more debug info to TFTP helper
+
+Hideaki Yoshifuji:
+  o [IPV6]: Mark MLDv2 report as known
+  o [IPV6]: Use IANA icmpv6 type for MLDv2 report
+
+Jakub Bogusz:
+  o rivafb 16bpp text background colour fix
+
+Jamal Hadi Salim:
+  o [NET_SCHED]: Check for NULL opt in dsmark_init
+
+Jeff Garzik:
+  o [TG3]: Dump NIC-specific statistics via ethtool
+
+Jon Oberheide:
+  o [CRYPTO]: ARC4 config help clarification
+
+Marcel Holtmann:
+  o [Bluetooth] Allow normal users to release the previous created TTY
+  o [Bluetooth] Fix race in RX complete routine of the USB drivers
+  o [Bluetooth] Make use of request_firmware() for the 3Com driver
+  o [Bluetooth] Add UART protocol id's for 3-Wire and H4DS
+
+Marcelo Tosatti:
+  o Andrew Morton: __free_pages_ok() stress testing
+  o Changed EXTRAVERSION to -pre2
+
+Matt Porter:
+  o ppc32: fix head_44x.S copyrights
+
+Nathan Scott:
+  o [XFS] Fix a very hard-to-hit, small-block-size only corruption
+  o [XFS] Fix delayed write buffer handling to use the correct list interfaces, add validity checks, remove unused code, fix comments.
+  o [XFS] Make buffer error checking consistent, add a value range check
+  o [XFS] Return the right error code on an ACL xattr version mismatch
+  o [XFS] Allow xfsbufd flush intervals to take immediate effect after changing the flush sysctl value.  Fix from Bart Samwel.
+  o [XFS] Clear the superblock dirty flag after flushing the log in sync_super.
+  o [XFS] Fix vmtruncate abuse in the XFS setattr ATTR_SIZE operation
+  o [XFS] make return value type for read() really ssize_t
+  o [XFS] Define a new superblock field for more feature bits
+  o [XFS] Fix debug builds - need sb_features2 details in the endian translation code.
+  o Remove a bk ignored XFS cvs directory, accidentally added
+  o Fix typo in delayed allocate buffer count reporting (sysrq-m)
+  o Minor updates to XFS documentation
+  o [XFS] Fix fsync regression resulting from moving data flushing out from under the IOLOCK.
+  o [XFS] Remove extraneous vmtruncate call, missed in earlier merge
+  o [XFS] Remove xfs_iaccess checks on security extended attribute namespace, done outside the filesystem.
+
+Russell Cattelan:
+  o [XFS] Fix for the xfs dir2 rebalance bug
+
+Rusty Russell:
+  o [NETFILTER]: Missing ip_rt_put in ipt_MASQUERADE
+
+Sridhar Samudrala:
+  o [SCTP] Fix typo in entry name of the remove_proc_entry() call
+  o [SCTP] Update sctp_ulpevent structure to include assoc pointer and
+  o [SCTP] Avoid the use of constant SCTP_IP_OVERHEAD to determin the max data size in a SCTP packet.
+  o [SCTP] Cleanup sctp_packet and sctp_outq infrastructure
+  o [SCTP] Partial Reliability Extension support
+  o [SCTP] Propagate error from sctp_proc_init. (Olaf Kirch)
+
+Stephen C. Tweedie:
+  o fix O(N^2) dquot sync behaviour
+
+Takayoshi Kochi:
+  o PCI Hotplug: acpiphp unable to power off slots
+
+Takayoshi Kouchi:
+  o PCI Hotplug: acpiphp cleanup patch for 2.4.23-pre4
+
+William Lee Irwin III:
+  o Joel Becker: Fix summit crash: cpu_present_to_apicid() bound checking
+
+Zwane Mwaikambo:
+  o fix module load with gcc3.3.3
+  o Fix typo in include/linux/compiler.h
+
+
+Summary of changes from v2.4.26 to v2.4.27-pre1
+============================================
+
+<khawar.chaudhry:amd.com>:
+  o Update amd8111 net driver
+
+<kirillx:7ka.mipt.ru>:
+  o Fix potential memory leak in devpts
+  o Fix potential memory access to free memory in /proc handling
+
+<lkml:lievin.net>:
+  o tipar char driver (divide by zero)
+
+<maragato:gmx.net>:
+  o Add ATI IGP 345M rev2 ID's
+
+<mhf:linuxmail.org>:
+  o Update codingstyle to 2.6 level
+
+<sezero:superonline.com>:
+  o megaraid2 compilation fix
+
+<tmattox:engr.uky.edu>:
+  o [netdrvr tulip] add MII support for Comet chips
+
+<vda:port.imtp.ilyichevsk.odessa.ua>:
+  o gcc3 does not inline some functions
+
+Atul Mukker:
+  o megaraid2 driver version 2.10.3
+
+Chris Wright:
+  o e1000: fix probable security hole
+
+Don Fry:
+  o resync pcnet32.c with 2.6.x
+  o netdevice.h add netif_msg_init helper
+  o pcnet32 fix hang/crash with loopback test
+
+Ganesh Venkatesan:
+  o e100: NFS/TCO related Firmware update
+  o e100: change log + version update
+  o e100: use new API, SET_NETDEV_DEV, rx_bytes stat to include MAC header fix
+  o e1000: ethtool set/get ring param support
+  o e1000: backoff Tanacross missed interrupt workaround
+  o e1000: Changed E1000_COLLISION_THRESHOLD from 16 to 15
+  o e1000: use E1000_PBA_BYTES_SHIFT instead of E1000_TX_FIFO_SIZE
+  o e1000: remove polarity reversal workaround for forced 10H/10F links
+  o e1000: fix eeprom update to include e1000_standby_eeprom
+  o e1000: new bit definitions, fix comments
+  o e1000: ethtool set/get eeprom fixes
+  o e1000: use new API, SET_NETDEV_DEV, check register_netdev retval
+  o e1000: all other white space fixes, changelog
+  o e1000: Disable TSO - till TSO related Tx hangs are root caused/fixed
+  o e1000: msec_delay fix
+  o e1000: phy fix, and cleanup
+
+Jeff Garzik:
+  o [netdrvr 8139cp] trivial syncing with 2.6.x
+  o [netdrvr 8139cp] rearrange priv struct, add cacheline-align markers
+  o [netdrvr 8139cp] locking cleanups
+  o [NET] forward-compat definition of netdev_priv()
+  o [netdrvr 8139cp] minor cleanups
+  o [netdrvr 8139cp] use netdev_priv()
+  o [netdrvr 8139cp] complete 64-bit DMA (PCI DAC) support
+  o [netdrvr 8139cp] better dev->close() handling, and misc related stuff
+  o [netdrvr natsemi] correct DP83816 IntrHoldoff register offset
+  o [netdrvr tulip] remove ChangeLog file, we have BitKeeper logs now
+  o Add SATA support
+  o Add Promise SX8 SATA driver
+  o [netdrvr via-rhine] Fix MII phy scanning bug, whitespace cleanups
+
+Marcelo Tosatti:
+  o Herbert Xu: Delete unused drivers/sound/Hwmcode.h, drivers/sound/724hwmcode.h
+  o Cset exclude: loftin@ldl.fc.hp.com|ChangeSet|20040414205510|54931
+  o Changed EXTRAVERSION to 2.4.27-pre1
+  o Fix drmP.h fix typo
+
+Meelis Roos:
+  o Kaupo Arulo: only use set_max when it is present
+
+Paul Gortmaker:
+  o [netdrvr 8390] Fix 8390 log spam
+
+Pavel Roskin:
+  o Tulip endianess fix
+
+Pete Zaitcev:
+  o Improve USB printer locking
+  o More USB storage locking fixes
+
+Randy Dunlap:
+  o drmP.h doesn't need local cmpxchg() and __cmpxchg()
+
+Scott Feldman:
+  o Update MAINTAINERS with new e100/e100 maintainers
+
+Zwane Mwaikambo:
+  o fix module load with gcc3.3.3
+
