diff -urN kernel-source-2.4.26-benh/arch/ppc/boot/chrp/main.c kernel-source-2.4.26-chrp8mb/arch/ppc/boot/chrp/main.c
--- kernel-source-2.4.26-benh/arch/ppc/boot/chrp/main.c	2004-02-21 04:46:59.000000000 +0100
+++ kernel-source-2.4.26-chrp8mb/arch/ppc/boot/chrp/main.c	2004-07-01 12:54:30.901555488 +0200
@@ -34,7 +34,7 @@
 
 #define RAM_FREE	((unsigned long)(_end+0x1000)&~0xFFF)
 #define PROG_START	0x00010000
-#define PROG_SIZE	0x00400000 /* 4MB */
+#define PROG_SIZE	0x00800000 /* 8MB */
 
 #define SCRATCH_SIZE	(128 << 10)
 
@@ -69,7 +69,7 @@
 
     im = (char *)(&__image_begin);
     len = (char *)(&__image_end) - (char *)(&__image_begin);
-    /* claim 4MB starting at PROG_START */
+    /* claim 8MB starting at PROG_START */
     claim(PROG_START, PROG_SIZE - PROG_START, 0);
     dst = (void *) PROG_START;
     if (im[0] == 0x1f && im[1] == 0x8b) {
@@ -77,7 +77,7 @@
 	begin_avail = avail_high = avail_ram;
 	end_avail = scratch + sizeof(scratch);
 	printf("gunzipping (0x%p <- 0x%p:0x%p)...", dst, im, im+len);
-	gunzip(dst, 0x400000, im, &len);
+	gunzip(dst, PROG_SIZE  - PROG_START, im, &len);
 	printf("done %u bytes\n\r", len);
 	printf("%u bytes of heap consumed, max in use %u\n\r",
 	       avail_high - begin_avail, heap_max);
diff -urN kernel-source-2.4.26-benh/arch/ppc/boot/chrp/Makefile kernel-source-2.4.26-chrp8mb/arch/ppc/boot/chrp/Makefile
--- kernel-source-2.4.26-benh/arch/ppc/boot/chrp/Makefile	2004-02-21 04:46:59.000000000 +0100
+++ kernel-source-2.4.26-chrp8mb/arch/ppc/boot/chrp/Makefile	2004-07-01 12:54:30.996541048 +0200
@@ -8,7 +8,7 @@
 
 USE_STANDARD_AS_RULE := true
 
-LD_ARGS = -T ../ld.script -Ttext 0x00400000
+LD_ARGS = -T ../ld.script -Ttext 0x00800000
 
 OBJS = ../common/crt0.o start.o main.o misc.o ../common/string.o image.o \
 	../common/ofcommon.o
